src/typed_tuple.rs

```diff
@@ -342,15 +276,137 @@ pub trait TypedTuple<Idx: TupleIndex, T>:
     /// ```rust
     /// # use typed_tuple::prelude::*;
     /// let tuple = (1u8, 2u16, 3u32, 4u64, 5i8);
-    /// let (left, right) = TypedTuple::<TupleIndex2, u32>::split_inclusive(tuple);
+    /// let (left, right) = IndexedTuple::<TupleIndex2, u32>::split_inclusive(tuple);
     /// assert_eq!(left, (1u8, 2u16, 3u32));
     /// assert_eq!(right, (3u32, 4u64, 5i8));
     /// ```
-    fn split_inclusive(self) -> (Self::SplitLeftInclusive, Self::SplitRightInclusive)
+    fn split_inclusive<INDEX>(self) -> (Self::SplitLeftInclusive, Self::SplitRightInclusive)
     where
+        Self: IndexedTuple<INDEX, T>,
+        INDEX: TupleIndex,
         T: Clone,
     {
         let (left_exclusive, element, right_exclusive) = self.split_exclusive();
-        (left_exclusive.chain_right((element.clone(),)), right_exclusive.chain_left((element,)))
+        (
+            left_exclusive.chain_right((element.clone(),)),
+            right_exclusive.chain_left((element,)),
+        )
+    }
+
+    #[inline]
+    /// Pops the element of type `T` from the tuple, returning it along with
+    /// the remaining tuple.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # use typed_tuple::prelude::*;
+    /// let mut tuple = (1u8, 2u16, 3u32);
+    /// let (val, rest) = tuple.pop_at::<TupleIndex2>();
+    /// assert_eq!(val, 3u32);
+    /// assert_eq!(rest, (1u8, 2u16));
+    /// ```
+    #[deprecated = "Replaced by `IndexedTuple::pop`"]
```

Comment:
Why marking this method with deprecated instead of just removing it? These are not stable APIs